lib/gitlab/client/build_triggers.rb

```diff
@@ -21,7 +22,7 @@ def triggers(project)
     # @param  [Integer] project The ID of a project.
     # @param  [String] token The token of a trigger.
     # @return [Gitlab::ObjectifiedHash] The trigger.
-    def trigger(project, token)
+    def trigger_detail(project, token)
```

Comment:
I hear what you are saying.  I guess all things considered it made sense to me to match the pattern of other Gitlab objects.  Plus the inline doc you provided states its purpose:
```
# Gets details of project's build trigger.
```
`trigger_build` seems like a good name to me.